### PR TITLE
UGENE-7471 Add the Use ambiguous bases checkbox to In silico PCR

### PR DESCRIPTION
**Pull request by DmitriiSukhomlinov**

### CODE REVIEW

File: src/plugins/GUITestBase/src/GTUtilsPcr.cpp [hunk 1]
```diff
@@ -56,6 +57,11 @@ void GTUtilsPcr::setMaxProductSize(HI::GUITestOpStatus &os, int number) {
     GTSpinBox::setValue(os, spinBox, number, GTGlobals::UseKeyBoard);
 }
 
+void GTUtilsPcr::setUseAmbiguousBases(HI::GUITestOpStatus& os, bool useAmbiguousBases) {
+    auto checkBox = dynamic_cast<QCheckBox*>(GTWidget::findWidget(os, "useAmbiguousBasesCheckBox"));
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/GTUtilsPcr.cpp]: use GTWidget::findCheckBox  or check for null the result of 'dynamic_cast'

File: src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp [hunk 2]
```diff
@@ -668,5 +672,101 @@ GUI_TEST_CLASS_DEFINITION(test_0017) {
...
+    CHECK_SET_ERR(count == 0, QString("Unexpected results count, expected: 0, current: %1").arg(count));
+}
+
+GUI_TEST_CLASS_DEFINITION(test_0019) {
+    // 1. Open WD
+    GTUtilsWorkflowDesigner::openWorkflowDesigner(os);
+
+    class Scenario : public CustomScenario {
+    public:
+        void run(HI::GUITestOpStatus& os) {
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp]: add override

File: src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp [hunk 3]
```diff
@@ -668,5 +672,101 @@ GUI_TEST_CLASS_DEFINITION(test_0017) {
...
+    GTUtilsDialog::waitForDialog(os, new WizardFiller(os, "In Silico PCR", new Scenario()));
+    GTUtilsWorkflowDesigner::addSample(os, "In silico PCR");
+
+    // 8. Set output to sandBoxDir + "result.gb"
+    GTUtilsWorkflowDesigner::click(os, "Write Sequence");
+    GTUtilsWorkflowDesigner::setParameter(os, "Output file", sandBoxDir + "result.gb", GTUtilsWorkflowDesigner::valueType::lineEditWithFileSelector);
+
+    // 9. Run workflow
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+    GTUtilsWorkflowDesigner::runWorkflow(os);
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp:746]: wait until workflow is finished.

File: src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp [hunk 4]
```diff
@@ -668,5 +672,101 @@ GUI_TEST_CLASS_DEFINITION(test_0017) {
...
+
+    // 11. Set "Use ambiguous bases", change output to sandBoxDir + "result_1.gb"
+    GTUtilsWorkflowDesigner::click(os, "In Silico PCR");
+    GTUtilsWorkflowDesigner::setParameter(os, "Use ambiguous bases", false, GTUtilsWorkflowDesigner::valueType::comboValue);
+    GTUtilsWorkflowDesigner::click(os, "Write Sequence");
+    GTUtilsWorkflowDesigner::setParameter(os, "Output file", sandBoxDir + "result_1.gb", GTUtilsWorkflowDesigner::valueType::lineEditWithFileSelector);
+
+    // 12. Run workflow
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+    GTUtilsWorkflowDesigner::runWorkflow(os);
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp:763]: wait until workflow is finished

**Review by yalgaer — CHANGES_REQUESTED** [review]
Minor comments only. 
The only blocker is that the test does not wait until workflow is finished.

File: src/plugins/GUITestBase/src/GTUtilsPcr.cpp (hunk 1 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/GUITestBase/src/GTUtilsPcr.cpp]: OK

File: src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp (hunk 2 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp]: OK

File: src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp (hunk 3 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp:746]: I have waiting one line above

File: src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp (hunk 4 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp:763]: Same

**Comment by DmitriiSukhomlinov** [comment]
See comments, tests have been passed

File: src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp (hunk 3 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp:746]: You need to wait after you run the workflow (first start, next wait for the workflow to finish), not before.

File: src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp (hunk 4 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp:763]: Same

**Review by yalgaer — CHANGES_REQUESTED** [review]
Only 1 issue left and without it the test may be flaky: wait until workflow is finished before checking results.

**Review by smlnkcv — DISMISSED** [review]

File: src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp (hunk 4 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp:763]: OK

File: src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp (hunk 3 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/GUITestBase/src/tests/common_scenarios/pcr/GTTestsInSilicoPcr.cpp:746]: OK

**Review by yalgaer — APPROVED** [review]

**Pull request merged**